### PR TITLE
Generation commands and complete setup command

### PR DESCRIPTION
**Pull request by lukasbischof**

### CODE REVIEW

File: lib/micro_deploy/cli/project_generate_group.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  module CLI
+    class ProjectGenerateGroup < Thor::Group
+      REPOSITORY_DIRECTORY = 'repository.git'
+
+      include Thor::Actions
+
+      argument :name, type: :string, desc: 'The name of the project'
+
+      def setup_git
+        system! "git init --bare #{REPOSITORY_DIRECTORY}", "Initialize empty git repository at #{REPOSITORY_DIRECTORY}"
```
Comment (TheFehr) [inline lib/micro_deploy/cli/project_generate_group.rb:19]: Should we add a check that git is installed before calling it?

File: lib/micro_deploy/cli/templates/hooks/pre-receive [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+ARGF.each do |line|
+  _, newrev, refname = line.split(' ')
+  next unless refname == 'refs/heads/master'
+
+  Dir.mktmpdir("#{project}_#{environment}") do |tmp_dir|
+    git_archive_command = DSL::Command.new(%W[git archive #{newrev}])
+    tar_command = DSL::Command.new(%W[tar -x -C #{tmp_dir}])
+
+    git_archive_command.pipe(tar_command).execute
```
Comment (TheFehr) [inline lib/micro_deploy/cli/templates/hooks/pre-receive:26]: And here as well. This hook will be broken if there is no git. So probably check it

File: lib/micro_deploy/cli/project_generate_group.rb (hunk 1 above)
Comment (lukasbischof) [reply lib/micro_deploy/cli/project_generate_group.rb:19]: Yes why not?

**Review by TheFehr — COMMENTED** [review]
Check git installed

File: lib/micro_deploy/cli.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+module MicroDeploy
+  module CLI
```
Comment (lukasbischof) [inline lib/micro_deploy/cli.rb]: bu

**Review by TheFehr — APPROVED** [review]
Add git check in another pull request

**Pull request merged**